src/typeOf/typeOf.spec.ts

```diff
@@ -0,0 +1,200 @@
+import { typeOf } from './typeOf';
+
+describe('typeOf function', () => {
+  const input = [1, 2, 3, 4, 5];
+
+  test('is a pure function', () => {
+    typeOf(input);
+    expect(input).toMatchSnapshot();
+  });
+
+  test('1 |> typeOf === "number"', () => {
+    expect(typeOf(1)).toBe('number');
+  });
+  test('undefined |> typeOf === "undefined"', () => {
+    expect(typeOf(undefined)).toBe('undefined');
+  });
+  test('null |> typeOf === "null"', () => {
+    expect(typeOf(null)).toBe('null');
+  });
+  test('NaN |> typeOf === "number"', () => {
+    expect(typeOf(NaN)).toBe('number');
+  });
+  test('Infinity |> typeOf === "number"', () => {
+    expect(typeOf(Infinity)).toBe('number');
+  });
+  test('5 |> typeOf === "number"', () => {
+    expect(typeOf(5)).toBe('number');
+  });
+  test('{} |> typeOf === "object"', () => {
+    expect(typeOf({})).toBe('object');
+  });
+  test('{ a: 1 } |> typeOf === "object"', () => {
+    expect(typeOf({ a: 1 })).toBe('object');
+  });
+  test('new Class() |> typeOf === "object"', () => {
+    class Foo {}
+    expect(typeOf(new Foo())).toBe('object');
+  });
+  test('function () {} |> typeOf === "function"', () => {
```

Comment:
There is a problem with the label for this test. It should be _new function(){}() |> typeOf === "object"_